### PR TITLE
ERC1155 Gating in client + E2E testing

### PR DESCRIPTION
**Pull request by clemire**
Additionally, entitlement tests are split into backwards compatible and v2-only entitlement tests are v2 tests are conditionally run in CI for v2 spaces only.

### CODE REVIEW

File: packages/sdk/src/channelsWithEntitlements.test.ts [hunk 1]
```diff
@@ -36,178 +37,16 @@ import {
...
-import { Client } from './client'
 import { make_MemberPayload_KeySolicitation } from './types'
 
 const log = dlog('csb:test:channelsWithEntitlements')
 const twoEth = BigInt(2e18)
 const oneEth = BigInt(1e18)
 const oneHalfEth = BigInt(5e17)
 
-// pass in users as 'alice', 'bob', 'carol' - b/c their wallets are created here
-async function setupChannelWithCustomRole(
```
Comment (clemire) [inline packages/sdk/src/channelsWithEntitlements.test.ts:52]: This is now shared with the channelsWithV2Entitlements test suite and was moved to util.test.ts

File: packages/sdk/src/spaceWithEntitlements.test.ts [hunk 2]
```diff
@@ -3,162 +3,45 @@
...
-    encodeRuleDataV2,
     encodeThresholdParams,
     createExternalNFTStruct,
 } from '@river-build/web3'
 
 const log = dlog('csb:test:spaceWithEntitlements')
 
-// Users need to be mapped from 'alice', 'bob', etc to their wallet addresses,
-// because the wallets are created within this helper method.
-async function createTownWithRequirements(requirements: {
```
Comment (clemire) [inline packages/sdk/src/spaceWithEntitlements.test.ts:54]: This method is now shared with the spaceWithV2Entitlements test suite and was moved to util.test.ts.

File: packages/sdk/src/util.test.ts [hunk 3]
```diff
@@ -953,3 +970,267 @@ export function isEncryptedData(obj: unknown): obj is EncryptedData {
         (typeof data.refEventId === 'string' || data.refEventId === undefined)
     )
 }
+// Users need to be mapped from 'alice', 'bob', etc to their wallet addresses,
+// because the wallets are created within this helper method.
+export async function createTownWithRequirements(requirements: {
```
Comment (clemire) [inline packages/sdk/src/util.test.ts:975]: copied from another file

File: packages/sdk/src/util.test.ts [hunk 4]
```diff
@@ -953,3 +970,267 @@ export function isEncryptedData(obj: unknown): obj is EncryptedData {
...
+        carolProvider,
+        alicesWallet,
+        bobsWallet,
+        carolsWallet,
+        spaceId,
+        channelId,
+        bobUserStreamView,
+    }
+}
+export async function expectUserCannotJoinSpace(
```
Comment (clemire) [inline packages/sdk/src/util.test.ts:1065]: copied over from another file

File: packages/sdk/src/channelsWithV2Entitlements.test.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
+/**
+ * @group with-v2-entitlements
```
Comment (clemire) [inline packages/sdk/src/channelsWithV2Entitlements.test.ts:2]: Here's a new test group for v2-only tests. Non-backwards-compatible erc1155 tests should only run on v2 spaces.

**Review by texuf — APPROVED** [review]

**Pull request merged**